numpy/core/src/multiarray/item_selection.c

```diff
@@ -1093,87 +1093,20 @@ _new_argsortlike(PyArrayObject *op, int axis, PyArray_ArgSortFunc *argsort,
 }
 
 
-/* Be sure to save this global_compare when necessary */
-static PyArrayObject *global_obj;
-
-static int
-sortCompare (const void *a, const void *b)
-{
-    return PyArray_DESCR(global_obj)->f->compare(a,b,global_obj);
-}
-
-/*
- * Consumes reference to ap (op gets it) op contains a version of
- * the array with axes swapped if local variable axis is not the
- * last dimension.  Origin must be defined locally.
- */
-#define SWAPAXES(op, ap) { \
-        orign = PyArray_NDIM(ap)-1; \
-        if (axis != orign) { \
-            (op) = (PyArrayObject *)PyArray_SwapAxes((ap), axis, orign); \
-            Py_DECREF((ap)); \
-            if ((op) == NULL) return NULL; \
-        } \
-        else (op) = (ap); \
-    }
-
-/*
- * Consumes reference to ap (op gets it) origin must be previously
- * defined locally.  SWAPAXES must have been called previously.
- * op contains the swapped version of the array.
- */
-#define SWAPBACK(op, ap) {                                      \
-        if (axis != orign) {                                    \
-            (op) = (PyArrayObject *)PyArray_SwapAxes((ap), axis, orign); \
-            Py_DECREF((ap));                                    \
-            if ((op) == NULL) return NULL;                      \
-        }                                                       \
-        else (op) = (ap);                                       \
-    }
-
-/* These swap axes in-place if necessary */
-#define SWAPINTP(a,b) {npy_intp c; c=(a); (a) = (b); (b) = c;}
-#define SWAPAXES2(ap) { \
-        orign = PyArray_NDIM(ap)-1; \
-        if (axis != orign) { \
-            SWAPINTP(PyArray_DIMS(ap)[axis], PyArray_DIMS(ap)[orign]); \
-            SWAPINTP(PyArray_STRIDES(ap)[axis], PyArray_STRIDES(ap)[orign]); \
-            PyArray_UpdateFlags(ap, NPY_ARRAY_C_CONTIGUOUS | \
-                                    NPY_ARRAY_F_CONTIGUOUS); \
-        } \
-    }
-
-#define SWAPBACK2(ap) { \
-        if (axis != orign) { \
-            SWAPINTP(PyArray_DIMS(ap)[axis], PyArray_DIMS(ap)[orign]); \
-            SWAPINTP(PyArray_STRIDES(ap)[axis], PyArray_STRIDES(ap)[orign]); \
-            PyArray_UpdateFlags(ap, NPY_ARRAY_C_CONTIGUOUS | \
-                                    NPY_ARRAY_F_CONTIGUOUS); \
-        } \
-    }
-
 /*NUMPY_API
  * Sort an array in-place
  */
 NPY_NO_EXPORT int
 PyArray_Sort(PyArrayObject *op, int axis, NPY_SORTKIND which)
 {
-    PyArrayObject *ap = NULL, *store_arr = NULL;
-    char *ip;
-    npy_intp i, n, m;
-    int elsize, orign;
-    int res = 0;
+    PyArray_SortFunc *sort;
     int axis_orig = axis;
-    int (*sort)(void *, size_t, size_t, npy_comparator);
+    int  n = PyArray_NDIM(op);
 
-    n = PyArray_NDIM(op);
```

Comment:
It is handled in `_new_sort_like`, anything with size <= 1 is left unchanged. So we delay the special casing a little.